packages/payment-detection/src/meta-payment-detector.ts

```diff
@@ -0,0 +1,187 @@
+import {
+  AdvancedLogicTypes,
+  CurrencyTypes,
+  ExtensionTypes,
+  PaymentTypes,
+  RequestLogicTypes,
+} from '@requestnetwork/types';
+import { deepCopy } from '@requestnetwork/utils';
+import { AnyToERC20PaymentDetector, AnyToEthFeeProxyPaymentDetector } from './any';
+import {
+  IPaymentNetworkModuleByType,
+  PaymentNetworkOptions,
+  ReferenceBasedDetectorOptions,
+} from './types';
+import { DeclarativePaymentDetector, DeclarativePaymentDetectorBase } from './declarative';
+import { BigNumber } from 'ethers';
+
+const supportedPns = [
```

Comment:
We could enforce type on this to get a linting error when we allow new extensions.
```suggestion
const supportedPns: (keyof ExtensionTypes.PnMeta.ICreationParameters)[] = [
```